core/signer/signer.go

```diff
@@ -66,56 +67,46 @@ func New(realm string,
 // StartSigning advance epochs once per minInterval, if there were mutations,
 // and at least once per maxElapsed minIntervals.
 func (s *Signer) StartSigning(ctx context.Context, minInterval, maxInterval time.Duration) {
-	// Fetch last time from previous map head (as stored in the map server) if it
-	// exists:
-	var last time.Time
+	var rootResp *trillian.GetSignedMapRootResponse
 	rootResp, err := s.tmap.GetSignedMapRoot(ctx, &trillian.GetSignedMapRootRequest{
 		MapId: s.mapID,
 	})
 	if err != nil {
 		glog.Infof("GetSignedMapRoot failed: %v", err)
-		last = time.Now()
-	} else {
-		mapRoot := rootResp.GetMapRoot()
-		if mapRoot != nil {
-			last = time.Unix(0, mapRoot.TimestampNanos)
-		} else {
-			last = time.Now()
+		// Immediately create new epoch and write new sth:
+		if err := s.CreateEpoch(ctx, true); err != nil {
+			glog.Fatalf("CreateEpoch failed: %v", err)
+		}
+		// Request map head again to get the exact time it was created:
+		rootResp, err = s.tmap.GetSignedMapRoot(ctx, &trillian.GetSignedMapRootRequest{
+			MapId: s.mapID,
+		})
+		if err != nil {
+			glog.Fatalf("GetSignedMapRoot failed after CreateEpoch: %v", err)
 		}
 	}
+	// Fetch last time from previous map head (as stored in the map server)
+	mapRoot := rootResp.GetMapRoot()
+	last := time.Unix(0, mapRoot.TimestampNanos)
```

Comment:
For protobuf objects, let's use a best practice of using the Get* operations. This will save us from a nil pointer dereference if we ever get one.